### PR TITLE
Site Assembler: Open Header, Sections, and Footer navigation to a side panel

### PR DESCRIPTION
**Pull request by miksansegundo**
_(edited on Aug 9 by @fushar)_

Related to:

- #78527

## Proposed Changes

* Show header, section, and footer patterns in a side panel
* Use an active state on the header/sections/footer buttons (same as the open category state)

### Before

https://github.com/Automattic/wp-calypso/assets/1525580/8616db67-5c92-4dc3-a4bc-d57bb2897718

### After

https://github.com/Automattic/wp-calypso/assets/1525580/6a3671c3-8574-4e44-bcea-48b83463bddd

## Testing Instructions

1. Access Site Assembler `/setup/with-theme-assembler/patternAssembler?siteSlug={ YOUR SITE }`
1. Click on "Header" and choose a header
1. Click on "Sections" and choose patterns (verify that the first selected category is All)
1. Click on "Footer" and choose a footer
1. Verify the active state of the header/sections/footer button
1. Verify the canvas scrolls to the header/footer area when the panel opens
1. Verify the selected header/section patterns/footer has a dark border on the pattern list (you'll have to move the focus from it)
1. Verify the hover, focus, and checked states of the header/sections/footer button on the main screen look as before
1. Verify that tracks events are not affected

## Pre-merge Checklist

- [X] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [X] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [X] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [X] Have you checked for TypeScript, React or other console errors?
- [X] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [X] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [X] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by taipeicoder** [comment]
I noticed that the "No header" "No footer" options are gone. 
Screenshot 2023-07-06 at 12 02 17 PM

**Comment by miksansegundo** [comment]
> I noticed that the "No header" "No footer" options are gone.

Yes, this is intentional for a more uniform UX within the assembler and the editor. Meaning that users will have to hover over the pattern on the canvas to see the button to remove it. I believe this will be their natural action.

Screenshot 2566-07-06 at 11 28 12

BTW. In the editor, to button says `Delete`. Let's update that copy in the assembler.
 

Also, @lucasmendes-design mentioned that we already made the decision to remove the pattern list screen, which has the pattern action bar to order, replace, and remove patterns. Meaning that users will have to arrange section patterns on the canvas as well. I can see how the whole experience is becoming more uniform.

File: client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx
Comment (taipeicoder) [inline client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx:419]: We still need to track when users click the back button in the Homepage panel. I noticed that the event was no longer firing. 
Screenshot 2023-07-06 at 12 57 34 PM

**Review by taipeicoder — APPROVED** [review]
Code LGTM! Let's wait for @lucasmendes-design feedback 🙂

File: client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx
Comment (miksansegundo) [reply client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx:419]: Thanks! I just added it to both pattern selection ("Sections") screens:

Screenshot 2566-07-06 at 12 44 12
Screenshot 2566-07-06 at 12 43 55

**Review by fushar — APPROVED** [review]
Nice UX!

**Comment by arthur791004** [comment]
The current implementation doesn't seem to get the benefit of NavigatorScreen. Hence, we can consider to get rid of it in the Assembler.

But I guess it would be simple if we update the usage of the NavigatorScreen to be the panel on the right side, so all of the sub-screens would be a side panel and we don't need the changes from https://github.com/Automattic/wp-calypso/pull/79066. For example, the initial screen renders `null`, and the others render the original panel as before including the pattern list (yep, we can make the pattern list as a screen now).

WDYT?

**Comment by lucasmendes-design** [comment]
Hi there, I tested and It's everything great :) 

> Yes, this is intentional for a more uniform UX within the assembler and the editor. Meaning that users will have to hover over the pattern on the canvas to see the button to remove it. I believe this will be their natural action.

I don't have strong opinions here. But it makes sense to me 👍🏼

**Comment by miksansegundo** [comment]
> The current implementation doesn't seem to get the benefit of NavigatorScreen. Hence, we can consider to get rid of it in the Assembler.

It was simpler for me to keep it because I see these PRs as quick prototypes to test the UX. 

> But I guess it would be simple if we update the usage of the NavigatorScreen to be the panel on the right side, so all of the sub-screens would be a side panel and we don't need the changes from https://github.com/Automattic/wp-calypso/pull/79066.

Thanks @arthur791004, I'll give it a try. Can you comment on https://github.com/Automattic/wp-calypso/pull/79066 to highlight the changes that we won't need?

**Comment by arthur791004** [comment]
> It was simpler for me to keep it because I see these PRs as quick prototypes to test the UX.
>
> But I guess it would be simple if we update the usage of the NavigatorScreen to be the panel on the right side, so all of the sub-screens would be a side panel and we don't need the changes from https://github.com/Automattic/wp-calypso/pull/79066.

Yeah, that makes sense! Now I think we can leave this PR as it is as something is still unsure (e.g.: The panel of the Colors & Fonts)

**Comment by autumnfjeld** [comment]
I tested this:

https://github.com/Automattic/wp-calypso/assets/5665959/e9287d0d-f3c6-4de6-bb96-845b4a0ef95e

Let's make some changes:

The click action on Header, Sections, Footer should feel the same.   The click action should open the pattern-list drawer.  This means that for a click on "Sections" two things happens at once: the category list opens and the “All” pattern drawer opens. 

I'll discuss with @taipeicoder in Marvel planning today so we have clarity.

**Comment by fushar** [comment]
@Automattic/lego could you re-review this PR please? I've updated the description and video.

Some known issues that I need help with, in future PRs:

- [ ] The keyboard accessibility feature is a bit broken in the Sections sublist. When we click tab until the Sections button, then hit enter, we somehow cannot reach to the sublist via tab and/or arrow keys 😢  need help maybe from @arthur791004; as I'm not super familiar with the existing logic in the codebase.
- [ ] Entry animation for the Sections category list: maybe I'm stupid but I somehow cannot make the animation works 😂 😭 I'll consider this as an enhancement which can be grabbed by any of you!

Thanks!

**Comment by fushar** [comment]
@lucasmendes-design , could you also re-review the design? 🙇

**Comment by miksansegundo** [comment]
Thanks for pushing this forward! I think the known issues can be considered enhancements for follow-up PRS, even the accessibility issue.

Here is a recording of me using this UX:

https://github.com/Automattic/wp-calypso/assets/1881481/8fe87044-8abf-42e3-a13e-7cadb711492e

About the UX, I see a challenge when navigating to the next steps when the categories are visible.

**Challenge** The navigation from `Sections` to the next steps (`Footer` or `Styles`) is not intuitive because it requires scrolling to see those options. 

**Idea 1.** We could try to always show all the next steps to users by only making scrollable the container of the categories so users don't lose the reference of the next steps. See sketch.
  
Screenshot 2566-08-10 at 18 34 57

**Idea 2.** We could try to do something to highlight the `Sections` toggle feature, maybe animate the caret icon, or use another color or something else. When you realize that the `Sections` can be toggled to hide the categories, then is easier to find the next steps. See recording.

https://github.com/Automattic/wp-calypso/assets/1881481/01d83877-810f-485a-b436-0f3de8bd2312

**Idea 3.** We could try to use the main button `Save and continue` to toggle the sections if they are open, and open the next uncompleted step (`Footer`) to guide users to see the next steps at least once before completing the flow and redirecting to the editor

What do you think about these ideas? 

I'm looking forward to getting feedback and more ideas to improve the navigation on an internal CfT.

**Comment by fushar** [comment]
Hi @miksansegundo, thanks for your review!

> Challenge The navigation from Sections to the next steps (Footer or Styles) is not intuitive because it requires scrolling to see those options.

I see your point and yes I just realized that the category list is hiding the rest of the steps.

> Idea 1. We could try to always show all the next steps to users by only making scrollable the container of the categories so users don't lose the reference of the next steps.

I tried implementing Idea 1. See below:

image

This is still a rough mockup; I think it looks off 😅 

Last night, @lucasmendes-design just proposed another iteration on Site Assembler, where the Styles group (Colors and Fonts) are available only in the next step. See p1691701318365409-slack-C048CUFRGFQ. So when it is implemented, when we do nothing, the only thing that this category list hides is (fortunately) just Footer step.

> Idea 2. We could try to do something to highlight the Sections toggle feature, maybe animate the caret icon, or use another color or something else.

I cannot see any animation or another color in your recording, but I see your point. I think we can just use another color / effect when the Section list is in open state.

---

I'd defer to @lucasmendes-design. What do you think? I'd suggest implementing this (if at all) as enhancement / future PR 😄

**Comment by autumnfjeld** [comment]
Can someone update the description of this issue?  Remove the current screen recording as it isn't accurate and sections for Before / After.

**Comment by autumnfjeld** [comment]
> Challenge The navigation from Sections to the next steps (Footer or Styles) is not intuitive because it requires scrolling to see those options.

I don't think this is a problem.  User must scroll to see all patterns.  User also must scroll to see the category list.   

Just like Lucas's design shows: p1691754296485899/1691701318.365409-slack-C048CUFRGFQ

**Comment by autumnfjeld** [comment]
> Last night, @lucasmendes-design just proposed another iteration on Site Assembler, where the Styles group (Colors and Fonts) are available only in the next step. See p1691701318365409-slack-C048CUFRGFQ. So when it is implemented, when we do nothing, the only thing that this category list hides is (fortunately) just Footer step.

@Automattic/lego  This is correct. We are 100% moving to a two-step process.

The task list in this PT shows that, it is up-to-date: pbxlJb-4lf-p2

image

**Comment by autumnfjeld** [comment]
@miksansegundo  @taipeicoder I think this version is shippable!!! Smooth! :)  Nice work.


https://github.com/Automattic/wp-calypso/assets/5665959/b9e855c5-3f2b-409a-af0a-dec33258d930

**Comment by fushar** [comment]
> Can someone update the description of this issue? Remove the current screen recording as it isn't accurate and sections for Before / After.

Done!

> I don't think this is a problem. User must scroll to see all patterns. User also must scroll to see the category list.

> @miksansegundo @taipeicoder I think this version is shippable!!! Smooth! :) Nice work.

Cool, I'll ship this today, unless someone from @Automattic/lego objects :)

**Comment by lucasmendes-design** [comment]
Hi there, here are some reviews about the list: 


The alignment is strange here:
image

I shared a better version for this component Site Assembler i4: 
image

> I don't think this is a problem. Users must scroll to see all patterns. Users also must scroll to see the category list.

+1. Let's keep it the way it is with the scroll.

**Comment by miksansegundo** [comment]
Hi @lucasmendes-design, I've created a PR to address those CSS fixes. See https://github.com/Automattic/wp-calypso/pull/80727

**Comment by lucasmendes-design** [comment]
Thanks Miguel :)

**Pull request merged**
